### PR TITLE
New Resources: `azurerm_dev_test_lab` / `azurerm_dev_test_virtual_network`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR adds support for the DevTest Labs and DevTest Virtual Network resources.

There's a bug in the Azure API where the Resource Group name is being returned in lower-case (regardless of the input case) - so this PR currently ignores the casing on the Resource Group name field.

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Resource tests pass:

screenshot 2018-09-20 at 09 51 54

File: azurerm/data_source_dev_test_lab.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+func dataSourceArmDevTestLab() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmDevTestLabRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```
Comment (katbyte) [inline azurerm/data_source_dev_test_lab.go]: This could be validated

File: azurerm/resource_arm_dev_test_lab.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDevTestLabName(),
```
Comment (katbyte) [inline azurerm/resource_arm_dev_test_lab.go]: Would this be better suited to be in the `azure` package with the other validators?

**Comment by tombuildsstuff** [comment]
Data Source tests pass:

screenshot 2018-09-20 at 10 10 27

File: azurerm/resource_arm_dev_test_virtual_network.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDevTestVirtualNetworkName(),
```
Comment (katbyte) [inline azurerm/resource_arm_dev_test_virtual_network.go]: as above, might this be better in the `azure` package?

File: azurerm/resource_arm_dev_test_lab.go [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+			},
+
+			"location": locationSchema(),
+
+			// TODO: file a bug about this being returned in lower-case
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_dev_test_lab.go]: TODO: open a GH issue and add a link for this

**Pull request merged**